File: server-mock/src/main/java/com/belogrudov/servermock/controller/Promise.java
package com.belogrudov.servermock.controller;

import java.time.LocalDate;

import lombok.Builder;

@Builder
public record Promise(LocalDate date, String title, String description, boolean isFulfilled) {
    @Override
    public String toString() {
        return """
                {
                    "date":"%s",
                    "title":"%s",
                    "description":"%s",
                    "is_fulfilled":%b
                }"""
                .formatted(date, title, description, isFulfilled);
    }
}

File: frontend/lib/model/Promise.dart
class Promise {
  final String date;
  final String title;
  final String description;
  final bool isFulfilled;

  Promise({required this.date, required this.title, required this.description, required this.isFulfilled});

  factory Promise.fromJson(Map<String, dynamic> json) {
    return Promise(
      date: json['date'],
      title: json['title'],
      description: json['description'],
      isFulfilled: json['is_fulfilled'],
    );
  }

  @override
  String toString() {
    return '$date $title';
  }
}
